packages/jest-cli/src/TestRunner.js

```diff
@@ -238,10 +247,22 @@ class TestRunner {
         aggregatedResults.success =
           aggregatedResults.numFailedTests === 0 &&
           aggregatedResults.numRuntimeErrorTestSuites === 0;
-        if (reporter.onRunComplete) {
-          reporter.onRunComplete(config, aggregatedResults);
-        }
-        return aggregatedResults;
+
+        return this._hasteMap.matchFiles(/\.snap/).then(res => {
```

Comment:
right, you can pass anything into the cleanup method that you like, as long as the actual cleanup happens somewhere else :)